kep/numa-nodes-scoring/README.md

```diff
@@ -0,0 +1,286 @@
+# NUMANodes Score plugin
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Topology format](#topology-format)
+  - [Score plugin implementation details](#score-plugin-implementation-details)
+    - [Description of calculating required NUMA nodes](#description-of-calculating-required-numa-nodes)
+    - [Description of substracting resources](#description-of-substracting-resources)
+    - [Description of normalizing score](#description-of-normalizing-score)
+- [Use cases](#use-cases)
+- [Known limitations](#known-limitations)
+- [Test plans](#test-plans)
+- [Graduation criteria](#graduation-criteria)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature enablement and rollback](#feature-enablement-and-rollback)
+- [Implementation history](#implementation-history)
+<!-- /toc -->
+
+# Summary
+
+This document describes behaviour of the Scoring plugin that scores nodes based on how many NUMA nodes are required to run given pod.
+
+# Motivation
+
+Consuming resources from multiple NUMA nodes can cause significant performance degradation
+in latency-critical execution and high-throughput applications.
+Topology Manager assigns resources from least amount of NUMA nodes but the scheduler is unaware
+of different NUMA topologies. The best case scenario would be to schedule pod on the node that
+can satisfy resource requirements from least amount of NUMA nodes to minimize latency.
+
+## Goals
+
+- Make more optimal scheduling decisions that take NUMA topology into consideration.
+
+## Non-Goals
+
+- Change the PodSpec to allow requesting a maximum amount of NUMA nodes required to satisfy resource requirements.
+- This Proposal requires exposing NUMA topology information. `NodeResourceTopology` API will be used to retrieve it.
+  Similiar to `NodeResourceTopology` plugins.
+
+# Proposal
+
+A Score plugin would check how many NUMA nodes are required to run given pod. Proposed Score plugin will use [CRD][1]
+to get available resources on the worker nodes and identify which topology policy is enabled, similliar to `NodeResourceTopology` plugins.
+It will run for all Topology Manager.
```

Comment:
it will be very nice if you can highlight the differences with NodeResourceTopology score. This can serve an explanation why a separate plugin is the best approach, or at least we can have a discussion around this area

Reply:
👍 good point